sandboxnu/graduatenu
# Reorg yarn workspaces

# Description

## Changes
- Removes a bunch of no longer useful files and directories(cross-checked with @bugsalexander before removing).
- Moves `api`, `api-v2`, `common`, `frontend` to packages directory.
- Makes each of the above directories a workspace in our yarn workspace monorepo. 
- Adds a package.json file with required scripts in `common` and `api`.
- Updates root package.json scripts to use yarn workspaces and migrate all shell scripts to yarn scripts.
- Add babel to build the common package + any future package that needs to be built and isn't our frontend/backend(since those use webpack).
- common is now a private package. Code shared through common is now imported using `import { xyz} from @graduate/common`.
- Add eslint in the project root and remove eslint config from `api-v2` directory.
- Update CI/CD hooks to use yarn workspaces.
- Update frontend dockerfile to use yarn workspaces.

## Review
- I found it pretty difficult to review this PR on github(it takes forever for the files to load). 99% of the changes are files being moves so it's probably best to simply checkout the branch and see what the directory structure looks like.
- If you want to look at specific changes then the package.json files + Dockerfile(for frontend) + CI/CD files in .github directory are the ones that have had major changes.

Closes [Add-yarn-workspaces-and-reorganize-code](https://trello.com/c/JUCFXpLy/401-add-yarn-workspaces-and-reorganize-code)

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

- Run `yarn dev` to run the frontend with the old backend. Everything should work as expected.
- Run `yarn dev:v2` to run the frontend with the new backend. Visiting `/api/students` should take give a response from the new backend and visiting `/xyz` should display the frontend. The frontend isn't able to fetch/post data to the backend though as expected.
- CI passes which means the frontend and backend docker images build correctly

# Checklist:
